ios-apps-master/Beer_Diary/Beer_Diary/Add_Beer.swift

```diff
@@ -0,0 +1,256 @@
+//
+//  Add_Beer.swift
+//  Beer_Diary
+//
+//  Created by Piotr Mielcarzewicz on 13/11/16.
+//  Copyright © 2016 Piotr Mielcarzewicz. All rights reserved.
+//
+
+import UIKit
+
+struct BeerItem
+{
+    var name = ""
+    var brewery = ""
+    var style = ""
+    var country = ""
+    var thoughts = ""
+    var aroma = 0
+    var appearance = 0
+    var palate = 0
+    var taste = 0
+    var overall:Float = 0
+    var date = NSDate()
+}
+
+func saveData()
+{
+    var tname = [String]()
+    var tbrewery = [String]()
+    var tstyle = [String]()
+    var tcountry = [String]()
+    var tthoughts = [String]()
+    var taroma = [Int]()
+    var tappearance = [Int]()
+    var tpalate = [Int]()
+    var ttaste = [Int]()
+    var tscore = [Float]()
+    var tdate = [NSDate]()
+    
+    
+    for x in BeerList
+    {
+        tname.append(x.name)
+        tbrewery.append(x.brewery)
+        tstyle.append(x.style)
+        tcountry.append(x.country)
+        tthoughts.append(x.thoughts)
+        taroma.append(x.aroma)
+        tappearance.append(x.appearance)
+        tpalate.append(x.palate)
+        ttaste.append(x.taste)
+        tscore.append(x.overall)
+        tdate.append(x.date)
+    }
+    
+    UserDefaults.standard.set(tname, forKey: "name")
+    UserDefaults.standard.set(tbrewery, forKey: "brewery")
+    UserDefaults.standard.set(tstyle, forKey: "style")
+    UserDefaults.standard.set(tcountry, forKey: "country")
+    UserDefaults.standard.set(tthoughts, forKey: "thoughts")
+    UserDefaults.standard.set(taroma, forKey: "aroma")
+    UserDefaults.standard.set(tappearance, forKey: "appearance")
+    UserDefaults.standard.set(tpalate, forKey: "palate")
+    UserDefaults.standard.set(ttaste, forKey: "taste")
+    UserDefaults.standard.set(tscore, forKey: "score")
+    UserDefaults.standard.set(tdate, forKey: "date")
```

Comment:
User defaults can nest arrays and dictionaries, so it would be much better to just have a key "beers" and have an array of dictionaries `{ "name" : "1", "brewery": "2", ... }` and so on, because currently you can have problems with the arrays desynchronizing and having different counts. Then your whole database of beers is corrupted